Editor/Scripts/Analytics/MeshSyncObserver.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using UnityEditor;
+
+namespace Unity.MeshSync.Editor.Analytics {
+    
+
+    /// <summary>
+    /// Hacky way to restore observers to MeshSync servers in scene
+    /// </summary>
+    [InitializeOnLoad]
+    internal static class MeshSyncObserverStartUp {
+
+        private static void update() {
+
+            var array = UnityEngine.Object.FindObjectsOfType<BaseMeshSync>(includeInactive: true);
+
+            foreach (var server in array) {
+                if (server.getNumObservers == 0) {
+
+                    var observer = new MeshSyncObserver();
+                    server.Subscribe(observer);
+                }
+            }
+
+            if (array.Length > 0) {
+                EditorApplication.update -= update;
+            }
```

Comment:
What if the user has not added a server to the scene?

Reply:
Yeah actually, does this update check run every editor update while there is no server in the scene?

Reply:
If it is running every frame, `FindObjectsOfType` will have a noticeable performance impact for users with large scenes

Reply:
But if you run this code, after 2 minutes of using the editor, i.e. going into playmode and back to edit mode, this will be true since this timer does not reset. As Sindharta mentioned, what about tracking time since the last check to control the check frequency?

Reply:
That's honestly still going to be very problematic for users with large projects for the first two minutes after loading a project, and will be really confusing if there's a sudden change in performance for no apparent reason two minutes after opening a project.

Unity's own best practices say to eliminate it altogether in production code
![image](https://user-images.githubusercontent.com/68228653/194080310-755b50db-d134-4db5-bf76-67b8ec65ce29.png)

Reply:
Latest change now only runs the init code with FindObjectOfType once per assembly reload.

Reply:
It's possible I'm missing something in the logic, but my understanding of the `update` function is now:

1. Start up observer. 
2. `alreadySetup` is false so don't unsubscribe.
3. Call `FindObjectsOfType`. `success` is false.
4. If there is no `BaseMeshSync` in the scene, `success` statys false.
5. `success` is false, so I don't unsubscribe from the editor update for 2 minutes, so this will continue running and updating on editor ticks for 2 minutes.

Is that correct? If so it doesn't address the startup performance issues.